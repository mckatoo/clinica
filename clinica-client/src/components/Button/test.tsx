
import { render, screen } from 'utils/test-utils';

import { CalendarPlus } from '@styled-icons/boxicons-regular';

import Button from './';

describe('<Button />', () => {
  it('should render the medium size by default', () => {
    const { container } = render(<Button>To Schedule</Button>)

    expect(screen.getByRole('button', { name: /To Schedule/i })).toHaveStyle({
      height: '4rem',
      padding: '0.8rem 3.2rem',
      'font-size': '1.4rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small size', () => {
    render(<Button size="small">To Schedule</Button>)

    expect(screen.getByRole('button', { name: /To Schedule/i })).toHaveStyle({
      height: '3rem',
      'font-size': '1.2rem'
    })
  })

  it('should render the large size', () => {
    render(<Button size="large">To Schedule</Button>)

    expect(screen.getByRole('button', { name: /To Schedule/i })).toHaveStyle({
      height: '5rem',
      'font-size': '1.6rem',
      padding: '0.8rem 4.8rem'
    })
  })

  it('should render a fullWidth version', () => {
    render(<Button fullWidth>To Schedule</Button>)

    expect(screen.getByRole('button', { name: /To Schedule/i })).toHaveStyle({
      width: '100%'
    })
  })

  it('should render an icon version', () => {
    render(
      <Button icon={<CalendarPlus data-testid="icon" />}>To Schedule</Button>
    )

    expect(screen.getByText(/to schedule/i)).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render a minimal version', () => {
    render(
      <Button icon={<CalendarPlus data-testid="icon" />} minimal>
        To Schedule
      </Button>
    )

    expect(screen.getByRole('button', { name: /to schedule/i })).toHaveStyle({
      background: 'none',
      color: '#ED3237'
    })

    expect(screen.getByRole('button', { name: /to schedule/i })).toHaveStyleRule(
      'background',
      'none',
      {
        modifier: ':hover'
      }
    )
  })

  it('should render a disabled Button', () => {
    render(<Button disabled>To Schedule</Button>)

    expect(screen.getByRole('button', { name: /to schedule/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled'
      }
    )
  })

  it('should render Button as a link', () => {
    render(
      <Button as="a" href="/link">
        To Schedule
      </Button>
    )

    expect(screen.getByRole('link', { name: /to schedule/i })).toHaveAttribute(
      'href',
      '/link'
    )
  })
})
