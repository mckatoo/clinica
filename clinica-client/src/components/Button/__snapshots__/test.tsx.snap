// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium size by default 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient( 180deg,#B3181C 0%,#EF4549 50% );
  color: #EBFFFD;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c0:focus {
  outline: 1px dashed;
}

.c0:hover {
  background: linear-gradient(180deg,#B3281C 0%,#EF5549 50%);
}

<button
  class="c0"
>
  <span>
    To Schedule
  </span>
</button>
`;
